### PR TITLE
Poy UI

### PR DESCRIPTION
**Pull request by apexearth**

### CODE REVIEW

File: components/GradientButton.tsx [hunk 1]
```diff
@@ -1,38 +1,49 @@
...
+  const wrap = (children: ReactNodeLike) =>
+    href ? (
+      <a href={href} className={wrapClassName}>
+        {children}
+      </a>
+    ) : (
+      <div className={wrapClassName}>{children}</div>
+    );
+  return wrap(
+    <button
```
Comment (toniocodo) [inline components/GradientButton.tsx:36]: not sure about nesting a button in a `a` tag, aren't you getting an html error?

File: pages/proof-of-yield/[timestamp]/[strategy].tsx [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+import { ExternalLinkButton } from "../../../components/ExternalLinkButton";
+import LineBarChart from "../../../components/LineBarChart";
+import { fetchDailyYields } from "../../../queries/fetchDailyYields";
+import { formatEther } from "viem";
+import Image from "next/image";
+import { useQuery } from "react-query";
+import { twMerge } from "tailwind-merge";
+import { TimeUnit } from "chart.js";
+import Tooltip from "../../../components/proof-of-yield/Tooltip";
+import * as dn from "dnum";
```
Comment (toniocodo) [inline pages/proof-of-yield/[timestamp]/[strategy].tsx:32]: you can destructure imports to leverage tree-shaking [ref](https://github.com/bpierre/dnum?tab=readme-ov-file#tree-shaking)

File: pages/proof-of-yield/[timestamp]/[strategy].tsx [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+import { useElementSize } from "usehooks-ts";
+
+const overrideCss = "px-8 md:px-10 lg:px-10 xl:px-[8.375rem]";
+
+const sma = (days: number) => {
```
Comment (toniocodo) [inline pages/proof-of-yield/[timestamp]/[strategy].tsx:37]: nit: the `periods` closure seems a bit strange to me, but I may be missing the goal

File: pages/proof-of-yield/[timestamp]/[strategy].tsx [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+                        <div className="horizontal-loader" />
+                      </div>
+                    ) : (
+                      formatN(allocationN, isSmall ? 1 : 2)
+                    )}
+                    {allocation && (
+                      <span className="font-normal ml-2 text-origin-white/70">
+                        (
+                        {`${dn.format(
+                          dn.mul(dn.div(allocation, totalBalance, 18), 100),
```
Comment (toniocodo) [inline pages/proof-of-yield/[timestamp]/[strategy].tsx:211]: very neat 👍

**Review by toniocodo — APPROVED** [review]
just a few nits, nothing blocking, lgtm 👍

File: components/GradientButton.tsx (hunk 1 above)
Comment (apexearth) [reply components/GradientButton.tsx:36]: I saw it done elsewhere in the app so I did it here too. 😅

File: pages/proof-of-yield/[timestamp]/[strategy].tsx (hunk 3 above)
Comment (apexearth) [reply pages/proof-of-yield/[timestamp]/[strategy].tsx:37]: An SMA needs to know a history for the length of periods it cares about. So I call the function saying I'm going to track a ~14 period SMA. The returned function is then used to calc the SMA for one specific range of inputs.

```typescript
const sma10 = sma(2)

sma10(5) // 5
sma10(1) // 3
sma10(2) // 1.5
```

**Pull request merged**